vlasovsolver/cpu_trans_map_amr.cpp

```diff
@@ -178,6 +250,198 @@ void flagSpatialCellsForAmrCommunication(const dccrg::Dccrg<SpatialCell,dccrg::C
+
+   // First: y-direction
+   for (uint i=0; i<localPropagatedCells.size(); i++) {
+      CellID c = localPropagatedCells[i];
+      SpatialCell *ccell = mpiGrid[c];
+      if (!ccell) continue;
+      // Sets the flag to true for all cells for now
+      ccell->SpatialCell::parameters[CellParams::AMR_TRANSLATE_COMM_X] = true;
+
+      // Is the cell translated?
+      if (!do_translate_cell(ccell)) continue;
+
+      // Is the cell a non-sysboundary cell?
+      // (translated sysboundarycells are only included via the neighbourhood loop)
+      if (ccell->sysBoundaryFlag != sysboundarytype::NOT_SYSBOUNDARY) continue;
+
+      LocalSet_y.insert(c);
+      // y-translation
+      findNeighborhoodCells(mpiGrid, c, 1, 1, foundCells);
+      for (uint j=0; j<foundCells.size(); j++) {
+         CellID n = foundCells[j];
+         SpatialCell *ncell = mpiGrid[n];
+         if (!ncell) continue;
+         // Is the cell translated?
+         if (!do_translate_cell(ncell)) continue;
+         LocalSet_y.insert(n);
+      }
+   }
+
+   // Then: x-direction
+   cellIDloop.clear();
+   cellIDloop.assign(LocalSet_y.begin(), LocalSet_y.end());
+   //for (auto it=LocalSet_y.begin(); it!=LocalSet_y.end(); ++it) {
+   //for (uint i=0; i<localPropagatedCells.size(); i++) {
+   for (uint i=0; i<cellIDloop.size(); i++) {
+      CellID c = cellIDloop[i];
+      //CellID c = localPropagatedCells[i];
+      //CellID c = (*it);
+      SpatialCell *ccell = mpiGrid[c];
+      if (!ccell) continue;
+      // Sets the flag to true for all cells for now
+      ccell->SpatialCell::parameters[CellParams::AMR_TRANSLATE_COMM_X] = true;
+
+      // Is the cell translated?
+      if (!do_translate_cell(ccell)) continue;
+
+      // Is the cell a non-sysboundary cell?
+      // (translated sysboundarycells are only included via the neighbourhood loop)
+      if (ccell->sysBoundaryFlag != sysboundarytype::NOT_SYSBOUNDARY) continue;
+
+      LocalSet_x.insert(c);
+      // x-translation
+      findNeighborhoodCells(mpiGrid, c, 0, 1, foundCells);
+      for (uint j=0; j<foundCells.size(); j++) {
+         CellID n = foundCells[j];
+         SpatialCell *ncell = mpiGrid[n];
+         if (!ncell) continue;
+         // Is the cell translated?
+         if (!do_translate_cell(ncell)) continue;
+         LocalSet_x.insert(n);
+      }
+   }
+
+   // Last: z-direction
+   cellIDloop.clear();
+   cellIDloop.assign(LocalSet_x.begin(), LocalSet_x.end());
+   //for (auto it=LocalSet_x.begin(); it!=LocalSet_x.end(); ++it) {
+   //for (uint i=0; i<localPropagatedCells.size(); i++) {
+   for (uint i=0; i<cellIDloop.size(); i++) {
+      CellID c = cellIDloop[i];
+      //CellID c = localPropagatedCells[i];
+      //CellID c = (*it);
+      SpatialCell *ccell = mpiGrid[c];
+      if (!ccell) continue;
+      // Sets the flag to true for all cells for now
+      ccell->SpatialCell::parameters[CellParams::AMR_TRANSLATE_COMM_X] = true;
```

Comment:
And this neither. I's due to the z->x->y ordering of translation, ensuring valid data is written to only the correct cells. Otherwise we can get severe corner artefacts. It is a bit of a brainbender.